src/main/java/mekanism/common/content/gear/mekatool/ModuleVeinMiningUnit.java

```diff
@@ -49,30 +51,41 @@ public int getExcavationRange() {
         return excavationRange.get().getRange();
     }
 
-    public static Set<BlockPos> findPositions(BlockState state, BlockPos location, Level world, int maxRange) {
+    public static Set<BlockPos> findPositions(Level world, Set<BlockPos> initial, boolean extended, int maxRange) {
         Set<BlockPos> found = new LinkedHashSet<>();
         Set<BlockPos> openSet = new LinkedHashSet<>();
-        openSet.add(location);
-        Block startBlock = state.getBlock();
-        int maxCount = MekanismConfig.gear.disassemblerMiningCount.get() - 1;
+        openSet.addAll(initial);
+        int singleMaxCount = MekanismConfig.gear.disassemblerMiningCount.get() - 1;
+        int maxCount = singleMaxCount * initial.size();
+        int next = openSet.size();
+        int dist = 0;
         while (!openSet.isEmpty()) {
             BlockPos blockPos = openSet.iterator().next();
             found.add(blockPos);
             openSet.remove(blockPos);
             if (found.size() > maxCount) {
                 return found;
+            } else if (found.size() == next) {
+                dist++;
+                next += openSet.size();
             }
-            for (BlockPos pos : BlockPos.betweenClosed(blockPos.offset(-1, -1, -1), blockPos.offset(1, 1, 1))) {
-                //We can check contains as mutable
-                if (!found.contains(pos) && (maxRange == -1 || WorldUtils.distanceBetween(location, pos) <= maxRange)) {
-                    Optional<BlockState> blockState = WorldUtils.getBlockState(world, pos);
-                    if (blockState.isPresent() && startBlock == blockState.get().getBlock()) {
-                        //Make sure to add it as immutable
-                        if (openSet.add(pos.immutable())) {
-                            //Note: We do this for all blocks we find/attempt to mine, not just ones we do mine, as it is a bit simpler
-                            // and also represents those blocks getting checked by the vein mining for potentially being able to be mined
-                            Mekanism.packetHandler().sendToAllTracking(new PacketLightningRender(LightningPreset.TOOL_AOE, Objects.hash(blockPos, pos),
-                                  Vec3.atCenterOf(blockPos), Vec3.atCenterOf(pos), 10), world, blockPos);
+            BlockState state = world.getBlockState(blockPos);
```

Comment:
I feel like it is worth adding a comment above this lines along the lines of:
```java
//Note: We don't need to validate the position is loaded as we only add loaded positions to our iteration set
```
Though thinking about it... as it basically goes from our iteration set, do you think it would be worthwhile to change `openSet` to being a `Map<BlockPos, BlockState>` so that we avoid having to query the world a second time?

Reply:
We can't change openSet to a map because of the ordering, but we could add a map to hold that, but then again I'm not sure how much faster a map would be than world?

Reply:
I mean it would just be a LinkedHashMap? Which is what `LinkedHashSet`s use internally anyway. And it should be quite a bit faster than the world as looking up from world also requires looking up the chunk and a bunch of other boiler plate on the way down.

Reply:
So the vein mine util method also uses the state. Should we change all of them to maps so that it'll just get it the first time? It's actually querying it 3+ times currently

Reply:
Was vein mining scaffolding working before? If it was I'm thinking that the break logic is handling the state being broken or something different. For now I changed it all to maps, but I left the world getter for the state in the util method just in case.

Reply:
Just tested briefly and it seems like scaffolding is working perfectly. Not duping or anything, I'll check with the cached state too although I suppose it's not exactly definitive as we don't know how other mods are reacting to state changes.

Reply:
One guess is without the skipping empty check is that if it AOE breaks a torch that is on a block the torch will fall normally but also cost energy to break.

Reply:
I tested with the cached state and it also handled drops fine for scaffolding I think, found a different issue with excavation range that might be affecting it. As far as the torch taking energy, I think that if it was in the area and highlighted that the user was prepared to use that energy even if it could have been done for free, or with an open hand for that matter. But there still might be other things the state could affect. Back to the original vein mining we could also get some of these potential issue if you veined a torch which was holding up sand and had a torch on it. This might take some investigation if we actually want to get to the bottum of it.

Reply:
I fixed the excavation range issue and it is still handling drops fine for scaffolding, so I think maybe this will work to use the mapped states if we're ok with potential other things like maybe energy for torches if that's the case.

Reply:
I switched it to the mapped state, but I'll switch it back if you don't think it's worth the risk.

Reply:
I will have to look closer when I start doing testing of this PR

Reply:
On second thought I think this should be switched back to make sure that `BlockState#onDestroyedByPlayer` and the like fire for the correct block and state (as with stairs breaking one will cause the state to change of neighboring ones for cases like corners, and if any mods actually care about the state in their destroy methods then we will be "breaking" the wrong one). I also think that because of then switching back away from the cached state, that before calculating destroy energy and the like we should check if it is air and if it is just skip.

Reply:
Will do. Are we confident that they can't become unbreakable? Or should we also check for that?

Reply:
They shouldn't but it might be worth checking